pkg/util/reservation_test.go

```diff
@@ -258,3 +259,318 @@ func TestIsObjValidActiveReservation(t *testing.T) {
+		if expect != got {
+			t.Errorf("reservation condition not equal, expect %v, got %v", expect, got)
+		}
+		return
+	}
+	if len(expect.Status.Conditions) != len(got.Status.Conditions) {
+		t.Errorf("reservation condition not equal, expect len %v, got len %v", len(expect.Status.Conditions), len(got.Status.Conditions))
+		return
+	}
+	expectConditions := map[string]*schedulingv1alpha1.ReservationCondition{}
+	for i, condition := range expect.Status.Conditions {
+		expectConditions[string(condition.Type)] = &expect.Status.Conditions[i]
+	}
+	for _, condition := range got.Status.Conditions {
+		e, ok := expectConditions[string(condition.Type)]
+		if !ok {
+			t.Errorf("reservation condition not equal, got unexpect condition type %v", condition.Type)
+			continue
+		}
+		msg := "condition type " + string(condition.Type)
+		assert.Equal(t, e.Status, condition.Status, msg)
+		assert.Equal(t, e.Message, condition.Message, msg)
+		assert.Equal(t, e.Reason, condition.Reason, msg)
+	}
+}
+
+var _ cache.ResourceEventHandler = &fakePodHandler{}
+
+type fakePodHandler struct {
+	t *testing.T
+}
+
+func (f *fakePodHandler) OnAdd(obj interface{}) {
+	_, ok := obj.(*corev1.Pod)
+	if !ok {
+		f.t.Errorf("OnAdd got object %T, but not a pod", obj)
+	}
+}
+
+func (f *fakePodHandler) OnUpdate(oldObj, newObj interface{}) {
+	_, ok := oldObj.(*corev1.Pod)
+	if !ok {
+		f.t.Errorf("OnUpdate got old object %T, but not a pod", oldObj)
+	}
+	_, ok = newObj.(*corev1.Pod)
+	if !ok {
+		f.t.Errorf("OnUpdate got new object %T, but not a pod", newObj)
+	}
+}
+
+func (f *fakePodHandler) OnDelete(obj interface{}) {
+	_, ok := obj.(*corev1.Pod)
+	if !ok {
+		f.t.Errorf("OnDelete got object %T, but not a pod", obj)
+	}
+}
+
+func TestReservationToPodEventHandlerFuncs(t *testing.T) {
+	testReservation := &schedulingv1alpha1.Reservation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "reserve-pod-0",
+		},
+		Spec: schedulingv1alpha1.ReservationSpec{
+			Owners: []schedulingv1alpha1.ReservationOwner{
+				{
+					Object: &corev1.ObjectReference{
+						Kind: "Pod",
+						Name: "test-pod-0",
+					},
+				},
+			},
+			TTL: &metav1.Duration{Duration: 30 * time.Minute},
+		},
+		Status: schedulingv1alpha1.ReservationStatus{
+			Phase: schedulingv1alpha1.ReservationPending,
+			Conditions: []schedulingv1alpha1.ReservationCondition{
+				{
+					Type:               schedulingv1alpha1.ReservationConditionScheduled,
+					Status:             schedulingv1alpha1.ConditionStatusTrue,
+					Reason:             schedulingv1alpha1.ReasonReservationScheduled,
+					LastProbeTime:      metav1.Now(),
+					LastTransitionTime: metav1.Now(),
+				},
+			},
+		},
+	}
+	t.Run("test not panic", func(t *testing.T) {
+		h := &ReservationToPodEventHandlerFuncs{
+			FilterFunc: func(obj interface{}) bool {
+				return true
+			},
+			PodHandler: &fakePodHandler{
+				t: t,
+			},
+		}
+
+		h.OnAdd(testReservation)
+
+		h.OnUpdate(testReservation, testReservation)
+
+		h.OnDelete(testReservation)
```

Comment:
could also add a case for `cache.DeletedFinalStateUnknown`

Reply:
done